arraykit.c

```diff
@@ -132,6 +132,87 @@ name_filter(PyObject *Py_UNUSED(m), PyObject *n)
     return n;
 }
 
+
+static PyObject *
+shape_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // If array is ndim 1, return (shape[0], 1), else return shape.
+    // https://numpy.org/doc/stable/reference/c-api/array.html
+    // https://stackoverflow.com/questions/56182259/how-does-one-acces-numpy-multidimensionnal-array-in-c-extensions
+
+    PyObject *shape = PyTuple_New(2);
+    if (!shape) {
+        return NULL;
+    }
+
+    PyArrayObject *array;
+    array = (PyArrayObject *)a;
+
+    // get size at dimension 0, set in tuple position 0
+    int size0 = PyArray_DIM(array, 0);
+    PyObject *rows = PyLong_FromLong(size0);
+    PyTuple_SET_ITEM(shape, 0, rows);
+
+    // handle 1 dimensional array
+    if (PyArray_NDIM(array) == 1 ) {
+        PyObject *cols = PyLong_FromLong(1);
+        PyTuple_SET_ITEM(shape, 1, cols);
+        return shape;
+    }
+
+    // get size at dimension 1, set in tuple position 1
+    int size1 = PyArray_DIM(array, 1);
+    PyObject *cols = PyLong_FromLong(size1);
+    PyTuple_SET_ITEM(shape, 1, cols);
+    return shape;
+
+}
+
+
+static PyObject *
+column_2d_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // If array ndim is 1, reshape into ndim 2 with 1 column
+    // related example: https://github.com/RhysU/ar/blob/master/ar-python.cpp
+
+    PyArrayObject *array;
+    array = (PyArrayObject *)a;
```

Comment:
Behold: the modern wonders of C99!

```suggestion
    PyArrayObject *array = (PyArrayObject *)a;
```